src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/InputManager.java

```diff
@@ -14,11 +15,14 @@
     private final IUserInput userInput = UserInputProvider.getUserInput(InputType.Keyboard);
 
     private final IndicationPacket indicationPacket;
+    private final GearPacket gearPacket;
 
     public InputManager(VirtualFunctionBus virtualFunctionBus) {
         super(virtualFunctionBus);
         this.indicationPacket = new IndicationPacket(userInput);
+        this.gearPacket = new GearPacket(userInput);
         virtualFunctionBus.indicationPacket = this.indicationPacket;
+        virtualFunctionBus.gearPacket = this.gearPacket;
```

Comment:
A billentyűlenyomások során keletkező események kiváltásának idejében nem igazán vagyok biztos, de extrém esetben  inkonzisztens állapot állhat be. 
**Példa:** 
1) Valamilyen érték van a **gearPacket**-ben, amelyet kiolvas a loop-ban soron következő komponens és használja.
2) Akár e közben vagy a művelet végeztével de érkezik egy új váltás esemény, aminek hatására új érték jelenik meg a packet-ben. 
3) Akár az említett komponens, vagy akár egy másik komponens ha újra olvassa a packet-et még ugyanazon loop-on belül, akkor nem ugyanazon értéket kapja, mint az első hívás során.

Azt kellene tehát átgondolni, hogy a működés szempontjából okozhat-e ez problémát.

**Megoldási javaslat**:
Én az irányjelző esetében amolyan "snapshot"-okat hoztam létre, hogy egy loop-on belül ne legyen ilyen probléma.  (30.sor)